uctMaths/competition/compadmin.py

```diff
@@ -1261,7 +1263,13 @@ def generate_school_answer_sheets(request, school_list):
     with zipfile.ZipFile(output_stringIO, 'w') as zipf:
         for ischool in school_list:
             output_string=printer_answer_sheet(request, ischool)
-            zipf.writestr(get_answer_sheet_name(ischool), output_string.getvalue())
```

Comment:
The PDF is specific to this year's competition. How feasible is it to make this future proof, e.g, a PDF that does not specify the year, or allow a new PDF to be uploaded each year by the competition administrator rather than by a Hexagon developer? The teacher declaration will also fall away if the competition returns to UCT campus in 2023.

Reply:
The Register Confirmation PDF is in it's own folder in the repo as a pdf, so I think it can be changed by the administrator by just adding a new file. For the declaration, would it work to place the part where it is added to the pdf in a try, finally statement so that it is skipped if the file is deleted, which is presumably what would happen next year?

Reply:
I think we should look into providing a way to upload the Register Confirmation PDF through the admin interface. I don't think we want to change the git repo and redeploy the site every year just to change a PDF.

Alternatively it might be easier to generate the PDF with dates changed? I don't know what you think is easier Du Toit.